raiden/network/proxies/token_network.py

```diff
@@ -52,6 +52,17 @@
 
 log = structlog.get_logger(__name__)  # pylint: disable=invalid-name
 
+ParticipantDetails = Dict[
+    str,
+    typing.Union[
```

Comment:
~~This is the wrong type constructor, [Union](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#union-types) builds a sum and not product type, we want to use [Tuple](https://mypy.readthedocs.io/en/stable/builtin_types.html#built-in-types) here~~

This is a bad type IMO, we should just use a namedtuple or a dataclass, which have a limited number of fields and we can individually type them.

Reply:
> This is a bad type IMO

This is the type that the function was returning before my PR.

I can change it and make it a named tuple, but that's just additional changes not really related to the PR.